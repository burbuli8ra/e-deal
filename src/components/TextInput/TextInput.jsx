import React from 'react';
import Styled from './TextInput.style'

const TextInput = ({
  onBlur= () => {},
  onChange = () => {},
  onKeyPress = () => {},
  placeholder,
  value
}) => {
  return (
    <Styled.TextInput
      data-testid="text-input"
      onBlur={onBlur}
      onChange={onChange}
      onKeyPress={onKeyPress}
      placeholder={placeholder}
      type="text"
      value={value}
    />
  );
};

export default TextInput