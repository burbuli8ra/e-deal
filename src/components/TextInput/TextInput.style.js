import styled from '@emotion/styled';
import { colors } from 'theme';

const TextInput = styled.input`
  background-color: ${colors.white};
  border: none;
  border-radius: 4px;
  box-shadow: inset 0 0 0 1px ${colors.pastelBlue};
  color: ${colors.darkGrey};
  font-size: 16px;
  outline: none;
  padding: 10px;
  width: 100px;
  
  &:focus {
    box-shadow: inset 0 0 0 2px ${colors.darkPastelBlue};
  }
`;

export default { TextInput };