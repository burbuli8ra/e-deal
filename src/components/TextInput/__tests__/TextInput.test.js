import React from 'react';
import { render } from '@testing-library/react';
import TextInput from '../TextInput';

const setup = props => <TextInput {...props} />;

describe('TextInput component', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {};
    wrapper = render(setup(props));
  });

  test('should match snapshot', () => {
    const { asFragment } = wrapper;
    expect(asFragment()).toMatchSnapshot();
  });
});