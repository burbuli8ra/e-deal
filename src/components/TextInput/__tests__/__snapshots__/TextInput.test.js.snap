// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput component should match snapshot 1`] = `
<DocumentFragment>
  <input
    class="css-1vb9rdv"
    data-testid="text-input"
    type="text"
    value=""
  />
</DocumentFragment>
`;
